src/www/app/outline_server_repository/index.ts

```diff
@@ -252,7 +282,15 @@ export class OutlineServerRepository implements ServerRepository {
   }
 
   private createServer(id: string, accessKey: string, name: string): OutlineServer {
-    const server = new OutlineServer(id, accessKey, name, this.createTunnel(id), this.net, this.eventQueue);
+    const server = new OutlineServer(
+      id,
+      accessKey,
+      name,
+      this.createTunnel(id),
+      this.net,
+      this.eventQueue,
+      this.isDynamicAccessKey(accessKey)
```

Comment:
Why should this be calculated in the constructor but the validation logic lives in the repository?

Reply:
Ah, I see, you need to calculate that in the repo.

Let's change that to a server type enum that is available in the model. We can have a Server.type, since we will expose that to the user.